src/main/kotlin/org/rust/lang/core/macros/MacroExpansionError.kt

```diff
@@ -8,44 +8,88 @@ package org.rust.lang.core.macros
 }
 
-sealed class MacroExpansionError {
-    data class Matching(val errors: List<MacroMatchingError>) : MacroExpansionError()
-    object DefSyntax : MacroExpansionError()
+sealed class MacroExpansionError
+
+sealed class DeclMacroExpansionError : MacroExpansionError() {
+    data class Matching(val errors: List<MacroMatchingError>) : DeclMacroExpansionError()
+    object DefSyntax : DeclMacroExpansionError()
 }
 
 sealed class MacroMatchingError(macroCallBody: PsiBuilder) {
     val offsetInCallBody: Int = macroCallBody.currentOffset
 
-    class PatternSyntax(macroCallBody: PsiBuilder): MacroMatchingError(macroCallBody) {
+    class PatternSyntax(macroCallBody: PsiBuilder) : MacroMatchingError(macroCallBody) {
         override fun toString(): String = "PatternSyntax"
     }
-    class ExtraInput(macroCallBody: PsiBuilder): MacroMatchingError(macroCallBody) {
+
+    class ExtraInput(macroCallBody: PsiBuilder) : MacroMatchingError(macroCallBody) {
         override fun toString(): String = "ExtraInput"
     }
-    class UnmatchedToken(macroCallBody: PsiBuilder, node: ASTNode): MacroMatchingError(macroCallBody) {
+
+    class UnmatchedToken(macroCallBody: PsiBuilder, node: ASTNode) : MacroMatchingError(macroCallBody) {
         val expectedToken: IElementType = node.elementType
         val expectedText: String = node.text
         val actualToken: IElementType? = macroCallBody.tokenType
         val actualText: String? = macroCallBody.tokenText
 
         override fun toString(): String = "UnmatchedToken($expectedToken(`$expectedText`) != $actualToken(`$actualText`))"
     }
-    class FragmentNotParsed(macroCallBody: PsiBuilder, val kind: FragmentKind): MacroMatchingError(macroCallBody) {
+
+    class FragmentNotParsed(macroCallBody: PsiBuilder, val kind: FragmentKind) : MacroMatchingError(macroCallBody) {
         override fun toString(): String = "FragmentNotParsed($kind)"
     }
-    class EmptyGroup(macroCallBody: PsiBuilder): MacroMatchingError(macroCallBody) {
+
+    class EmptyGroup(macroCallBody: PsiBuilder) : MacroMatchingError(macroCallBody) {
         override fun toString(): String = "EmptyGroup"
     }
-    class TooFewGroupElements(macroCallBody: PsiBuilder): MacroMatchingError(macroCallBody) {
+
+    class TooFewGroupElements(macroCallBody: PsiBuilder) : MacroMatchingError(macroCallBody) {
         override fun toString(): String = "TooFewGroupElements"
     }
-    class Nesting(macroCallBody: PsiBuilder, val variableName: String): MacroMatchingError(macroCallBody) {
+
+    class Nesting(macroCallBody: PsiBuilder, val variableName: String) : MacroMatchingError(macroCallBody) {
         override fun toString(): String = "Nesting($variableName)"
     }
 }
+
+sealed class ProcMacroExpansionError : MacroExpansionError() {
+    data class ServerSideError(val message: String) : ProcMacroExpansionError() {
+        override fun toString(): String = "ProcMacroExpansionError.Expansion(message = \"$message\")"
```

Comment:
```suggestion
        override fun toString(): String = "ProcMacroExpansionError.ServerSideError(message = \"$message\")"
```

Reply:
nit: Do you actually need `ProcMacroExpansionError.` prefix? If not, you can use `javaClass.simpleName` not to forget updating `toString`. If you need it, probably we can use something like `.javaClass.canonicalName.removePrefix("${javaClass.packageName}.")`.

Note, in fact you use `simpleName` for `DeclMacroExpansionError` errors.
Moreover, you can provide basic implementation of `toString` in this way and avoid `override fun toString(): String = "ProcMacroExpansionError.Timeout"` and friends